disnake/ext/commands/converter.py

```diff
@@ -355,27 +364,29 @@ def _get_id_matches(ctx, argument):
         data = match.groupdict()
         channel_id = disnake.utils._get_as_snowflake(data, "channel_id") or ctx.channel.id
         message_id = int(data["message_id"])
-        guild_id = data.get("guild_id")
-        if guild_id is None:
+        guild_id_str: Optional[str] = data.get("guild_id")
+        if guild_id_str is None:
             guild_id = ctx.guild and ctx.guild.id
-        elif guild_id == "@me":
+        elif guild_id_str == "@me":
             guild_id = None
         else:
-            guild_id = int(guild_id)
+            guild_id = int(guild_id_str)
         return guild_id, message_id, channel_id
 
     @staticmethod
-    def _resolve_channel(ctx, guild_id, channel_id) -> Optional[MessageableChannel]:
+    def _resolve_channel(
+        ctx: AnyContext, guild_id: Optional[int], channel_id: int
+    ) -> Optional[MessageableChannel]:
+        bot: disnake.Client = ctx.bot
         if guild_id is None:
-            return ctx.bot.get_channel(channel_id) if channel_id else ctx.channel
+            return bot.get_channel(channel_id) if channel_id else ctx.channel  # type: ignore
 
-        guild = ctx.bot.get_guild(guild_id)
-        if guild is not None and channel_id is not None:
+        guild = bot.get_guild(guild_id)
+        if guild is not None:
             return guild._resolve_channel(channel_id)  # type: ignore
```

Comment:
We probably don't need `# type: ignore` here anymore

Reply:
We do, since `_resolve_channel` returns `GuildChannel | ...` which includes `StageChannel` and hence won't work with `MessageableChannel`